pinot-core/src/main/java/org/apache/pinot/core/segment/index/loader/invertedindex/TextIndexHandler.java

```diff
@@ -161,16 +211,30 @@ private void createTextIndexForColumn(ColumnMetadata columnMetadata)
         for (int docId = 0; docId < numDocs; docId++) {
           textIndexCreator.addDoc(rawIndexReader.getString(docId, chunkReaderContext), docId);
         }
+        // TODO(pradeep): Handle non-dictionary case.
       } else {
+        LuceneFSTIndexCreator luceneFSTIndexCreator =
+                new LuceneFSTIndexCreator(
+                        _indexDir, column,
+                        columnMetadata.getCardinality(), null);
         // text index on dictionary encoded SV column
         // read forward index to get dictId
         // read the raw value from dictionary using dictId
         try (BaseImmutableDictionary dictionary = getDictionaryReader(columnMetadata)) {
+          for (int dictId = 0; dictId < dictionary.length(); dictId++) {
+            luceneFSTIndexCreator.addSortedDictIds(
```

Comment:
This will break the existing functionality provided and (used in Li) for text index - phrase/positional searches.

Reply:
Removed